### DISCUSSION TITLE
Editor can't "find" a variable when they contain special characters

### DISCUSSION THREAD

**Discussion by tehTK** [codefori/vscode-ibmi#1981] [Q&A]
Hi *ALL, 
Is there a setting in VScode that affects what counts as a variable and what not or is this a CodeForI thing ?

If we have variable names like " #DataStructureName " then the Editor can't find any Definitions or even have problems while coping with Ctrl-C/Ctrl-V.  The # is not counted as being part of the name. Therefore it cant find any definition for DataStrucutreName. This happens only for some special characters. For example § works fine. 

I cant't find any Setting that affects this so maybe someone has a similar problem since we have some code with # in variable names and they are valid RPG names.

**Comment by chrjorgensen** [comment]
@tehTK I don't have an answer for why variables are not found when they include special characters...

...but I would like you to consider why you have special characters and whether you should stop using those:

The US characters `#`, `@` and `$` are valid in variable names, but they are not consistent across different EBCDIC encodings and depends on the CCSID value. ILE RPG allows for long variable names, so there is no longer any reason for using those special characters.

I would strongly advice to stop using these characters and remove them from variable names in every source changed from now on. This will give you a much better developer experience in Code for IBM i, e.g. like better outline content and being able to rename the RPG variable (symbol) name by pressing F2 in the name.
